core_backend/app/user_tools/routers.py

```diff
@@ -32,45 +42,33 @@
 logger = setup_logger()
 
 
-@router.post("/", response_model=UserCreate)
+@router.post("/", response_model=UserCreateWithCode)
 async def create_user(
     user: UserCreateWithPassword,
     admin_user_db: Annotated[UserDB, Depends(get_admin_user)],
     request: Request,
     asession: AsyncSession = Depends(get_async_session),
-) -> UserCreate | None:
+) -> UserCreateWithCode | None:
     """
     Create user endpoint. Can only be used by admin users.
     """
 
     try:
```

Comment:
can we explicitly check if user already exists instead of using `try/except` logic?